esFrontLine/connection.py

```diff
@@ -56,4 +57,33 @@ def perform_request(self, method, url, params, body, headers=None, *args, **kwar
         hawk_headers = headers and headers.copy() or {}
         hawk_headers['Authorization'] = sender.request_header
 
-        return super(HawkConnection, self).perform_request(method, url, params, body, headers=hawk_headers, *args, **kwargs)
+        return hawk_headers
+
+
+class HawkConnection(BaseHawkConnection, Urllib3HttpConnection):
+    '''
+    Connection class to use with an elasticsearch synchronous python client
+    '''
+
+    def perform_request(self, method, url, params, body, headers=None, *args, **kwargs):
+        '''
+        Build a new HAWK header on each request
+        Using elasticsearch Synchronous client
+        '''
+        hawk_headers = self.add_hawk_authentication(method, url, params, body, headers)
+        return super(Urllib3HttpConnection, self).perform_request(method, url, params, body, headers=hawk_headers, *args, **kwargs)
+
+
+if ASYNC_SUPPORT:
```

Comment:
Personally, I would find it cleaner if the import and class definition were bundled together so `ASYNC_SUPPORT` variable is not needed.  Maybe you plan for this variable to be used in more places? 

```python
try:
    from elasticsearch_async.connection import AIOHttpConnection
    class AsyncHawkConnection(BaseHawkConnection, AIOHttpConnection):
```

No need to change anything.

Reply:
Better yet, put it in its own python file, and import it only if required.

Reply:
For the sake of compatibility with python2, which is still around, put the asynch class in its own python file.